### ISSUE TITLE
A Unified 3D Human Motion Synthesis Model via Conditional Variational Auto-Encoder

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#127]
## TL; DR

- Two-path CVAE framework that can handle diverse (action-label-conditioned) *masked* motion tasks

## Motivation

- Tasks like motion prediction, completion, interpolation can all be regarded as *masked motion modeling* problem.

## Previous Work

- Motion prediction
- Motion completion and Interpolation

## Method

![image](https://user-images.githubusercontent.com/46308407/144050536-3d4d34d3-b344-4f2c-8003-428009334d0d.png)

### (1) Framework

- Two path: 1) Generation from *seen* motion 2) Reconstruction from *unseen* GT motion
- ResNet-based Convolutional Networks

### (2) Action-Adaptive Modulation
![image](https://user-images.githubusercontent.com/46308407/144050759-3fb98180-7d02-4987-81b0-ac3989ef278c.png)

- To be able to condition on predefined action class label, introduce learned adaptive modulation (normalization) that takes action label and time positional encoding as input and outputs beta and gamma parameters.

![image](https://user-images.githubusercontent.com/46308407/144050894-380c2367-03b3-4805-8ffa-7936afc8e83f.png)

### (3) Cross-Attention
![image](https://user-images.githubusercontent.com/46308407/144050958-3a54027e-5789-433f-abe1-72c3cf8590a6.png)

### (4) Training

- KL Regularization
![image](https://user-images.githubusercontent.com/46308407/144051053-e9b08381-7a55-4c49-b911-48ce61912791.png)
![image](https://user-images.githubusercontent.com/46308407/144051068-615e9651-8592-4acd-8412-2cb3005f4543.png)

- Matching Loss
![image](https://user-images.githubusercontent.com/46308407/144051141-068cea38-6ea4-4a16-8bc8-0c2db73bb83c.png)
(bottom path: reconstructive) <br>
![image](https://user-images.githubusercontent.com/46308407/144051170-bde33626-26d0-4219-8595-e2f1a2ad314d.png)
(top path: generative) <br>

- Adversarial Loss
![image](https://user-images.githubusercontent.com/46308407/144051228-5b2b7437-338c-4eea-b309-f75d688dca59.png)
![image](https://user-images.githubusercontent.com/46308407/144051261-2f986098-1ddd-4590-be0e-9fd5d43f24cc.png)
(Feature Matching + LSGAN objective) <br>

- Classification
![image](https://user-images.githubusercontent.com/46308407/144051280-fba1dd5e-5490-418a-8de5-ce6eace4514b.png)

- Overall
![image](https://user-images.githubusercontent.com/46308407/144051319-8d2903b1-bea5-4feb-bd05-6c773da5297b.png)

## Comments
